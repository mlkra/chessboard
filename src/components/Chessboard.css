.chessboard {
    height: 600px;
    width: 600px;
    display: flex;
    flex-direction: column;
}

.chessboard__row {
    display: flex;
    flex: 1;
}

.chessboard__square {
    flex: 1;
}

.chessboard__square--light {
    background-color: whitesmoke;
}

.chessboard__square--dark {
    background-color: burlywood;
}

.chessboard__piece {
    display: block;
    height: 100%;
    width: 100%;
}
