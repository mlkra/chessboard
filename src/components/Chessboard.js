import React from "react";
import "./Chessboard.css";
import { calculateBoardCoordsFromMouseEvent, convertBoardCoordsToCoordinateNotation, convertFENToInternalRepresentation } from "./chessboardUtils";
import pieces from "./pieces";

export default function Chessboard({ 
    fen = "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w - - 0 1",
    onSquareSelected = () => {},
}) {
    const board = convertFENToInternalRepresentation(fen);

    let draggedPiece;

    function handleMouseDown(event) {
        console.log("down");
        draggedPiece = event.target;
        const coords = calculateBoardCoordsFromMouseEvent(event)
        onSquareSelected(convertBoardCoordsToCoordinateNotation(coords))
        event.preventDefault();
    }

    function handleMouseMove(event) {
        console.log("move");
        if (draggedPiece) {
            draggedPiece.style.transform = `translate(${event.pageX - draggedPiece.x - draggedPiece.width / 2}px,${event.pageY - draggedPiece.y - draggedPiece.height / 2}px)`;
        }
        event.preventDefault();
    }

    function handleMouseUp(event) {
        draggedPiece = null;
        const coords = calculateBoardCoordsFromMouseEvent(event)
        onSquareSelected(convertBoardCoordsToCoordinateNotation(coords))
        event.preventDefault();
    }

    return (
        <div className="chessboard" onMouseDown={handleMouseDown} onMouseMove={handleMouseMove} onMouseUp={handleMouseUp}>
            {board.map((row, rowIndex) => <div key={rowIndex} className="chessboard__row">
                {row.map((square, columnIndex) => {
                    const color = getSquareColor(rowIndex, columnIndex);
                    const piece = getSquarePiece(square);
                    return (
                        <div key={columnIndex} className={`chessboard__square chessboard__square--${color}`}>
                            {piece}
                        </div>
                    );
                })}
            </div>)}
        </div>
    );
}

function getSquareColor(rowIndex, columnIndex) {
    let color = "dark";
    if ((rowIndex + columnIndex) % 2 === 0) {
        color = "light";
    }
    return color;
}

function getSquarePiece(square) {
    if (square) {
        return <img className="chessboard__piece" src={pieces[square]} draggable="false"></img>;
    }
}
