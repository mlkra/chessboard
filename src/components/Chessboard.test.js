import React from "react";
import renderer from "react-test-renderer";
import Chessboard from "./Chessboard";

describe("Chessboard", () => {
    it("renders correctly with default fen", () => {
        expect(renderer.create(<Chessboard />).toJSON()).toMatchSnapshot();
    });
    it("renders empty board", () => {
        expect(renderer.create(<Chessboard fen="8/8/8/8/8/8/8/8 w - - 0 1" />).toJSON()).toMatchSnapshot();
    });
    it("renders board full of black pawns", () => {
        expect(renderer.create(<Chessboard fen="pppppppp/pppppppp/pppppppp/pppppppp/pppppppp/pppppppp/pppppppp/pppppppp w - - 0 1" />).toJSON()).toMatchSnapshot();
    });
});
