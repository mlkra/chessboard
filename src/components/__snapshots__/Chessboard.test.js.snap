// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chessboard renders board full of black pawns 1`] = `
<div
  className="chessboard"
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;

exports[`Chessboard renders correctly with default fen 1`] = `
<div
  className="chessboard"
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--dark"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
    <div
      className="chessboard__square chessboard__square--light"
    >
      <img
        className="chessboard__piece"
        draggable="false"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;

exports[`Chessboard renders empty board 1`] = `
<div
  className="chessboard"
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
  </div>
  <div
    className="chessboard__row"
  >
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
    <div
      className="chessboard__square chessboard__square--dark"
    />
    <div
      className="chessboard__square chessboard__square--light"
    />
  </div>
</div>
`;
